src/braket/circuits/qubit.py

```diff
@@ -36,8 +36,10 @@ def __new__(cls, index: int):
             >>> Qubit(0)
             >>> Qubit(1)
         """
+        if not isinstance(index, int):
```

Comment:
What's our stance on if someone supplies say numpy.int ? Should we accept that and cast it to int ?